web3_proxy/src/frontend/admin.rs

```diff
@@ -33,6 +41,141 @@ use std::sync::Arc;
+        })?;
+    let user_address_bytes: Vec<u8> = user_address.clone().to_fixed_bytes().into();
+    let note: String = params
+        .get("note")
+        .ok_or_else(|| {
+            Web3ProxyError::BadRequest("Unable to find 'note' key in request".to_string())
+        })?
+        .parse::<String>()
+        .map_err(|_| {
+            Web3ProxyError::BadRequest("Unable to parse 'note' as a String".to_string())
+        })?;
+    // Get the amount from params
+    // Decimal::from_str
+    let amount: Decimal = params
+        .get("amount")
+        .ok_or_else(|| {
+            Web3ProxyError::BadRequest("Unable to get the amount key from the request".to_string())
+        })
+        .map(|x| Decimal::from_str(x))?
+        .or_else(|err| {
+            Err(Web3ProxyError::BadRequest(format!(
+                "Unable to parse amount from the request {:?}",
+                err
+            )))
+        })?;
+
+    let user_entry: user::Model = user::Entity::find()
+        .filter(user::Column::Address.eq(user_address_bytes.clone()))
+        .one(&db_conn)
+        .await?
+        .ok_or(Web3ProxyError::BadRequest(
+            "No user with this id found".to_string(),
+        ))?;
+
+    let increase_balance_receipt = admin_increase_balance_receipt::ActiveModel {
+        amount: sea_orm::Set(amount),
+        admin_id: sea_orm::Set(admin_entry.id),
+        deposit_to_user_id: sea_orm::Set(user_entry.id),
+        note: sea_orm::Set(note),
+        ..Default::default()
+    };
+    increase_balance_receipt.save(&db_conn).await?;
+
+    let mut out = HashMap::new();
+    out.insert(
+        "user",
+        serde_json::Value::String(format!("{:?}", user_address)),
+    );
+    out.insert("amount", serde_json::Value::String(amount.to_string()));
+
+    // Get the balance row
+    let balance_entry: balance::Model = balance::Entity::find()
+        .filter(balance::Column::UserId.eq(user_entry.id))
+        .one(&db_conn)
+        .await?
+        .context("User does not have a balance row")?;
+
+    // Finally make the user premium if balance is above 10$
+    let premium_user_tier = user_tier::Entity::find()
+        .filter(user_tier::Column::Title.eq("Premium"))
+        .one(&db_conn)
+        .await?
+        .context("Premium tier was not found!")?;
+    if balance_entry.available_balance + amount >= Decimal::new(10, 0)
+        && user_entry.user_tier_id != premium_user_tier.id
+    {
+        let balance_entry = balance_entry.into_active_model();
+        balance::Entity::insert(balance_entry)
+            .on_conflict(
+                OnConflict::new()
+                    .values([
+                        // (
+                        //     balance::Column::Id,
+                        //     Expr::col(balance::Column::Id).add(self.frontend_requests),
+                        // ),
+                        (
+                            balance::Column::AvailableBalance,
```

Comment:
@WyseNynja is this type correct? I was looking at the docs but couldn't figure out how to do a simple update with this syntax (thus instead it's doing "insert on conflict", which will always result in a conflict)

Reply:
I was checking the active model functionality, couldn't find anything either. If you stumble across something please do let me know.